### PR TITLE
Refactor/component folders

### PR DESCRIPTION
**Pull request by teesloane**
Here's a big ol' PR with many things that have happened.

**PR Purpose**
- the original goal was to refactor a single component into a seperate folder to demo an MVP of seperated code components.
- This sort of fell apart as things started failing by nature of seperating out the code. 

**Main changes**
- Updating webpack to handle imports / css
-Moving components into their own files
- converting component styles from JSS -> CSS
- converting redux / actions / constants into seperated folders
- create a demo for showing a state change / action dispatch jam.

**Broken Things**
- Tests. They are borked.
- `SIDEPANEL` styling - not sure if I missed some style rules... seems like it. 
- Probably other things.

feedback welcome! There are plenty changes to be made I'm sure - thought it'd be about time to toss a PR up.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,5 +56,11 @@
     "redux": "^3.6.0",
     "socket.io-client": "^1.5.1",
     "webpack": "^1.14.0"
+  },
```
Comment (mihok) [inline package.json]: `style-loader`, and `css-loader` is missing from the devDependencies

File: src/components/Input/Input_styles.css [hunk 2]
```diff
@@ -0,0 +1,56 @@
+.Input_MESSENGER {
```
Comment (mihok) [inline src/components/Input/Input_styles.css]: This is driving my OCD nuts, can we call these files `Input.css`? to add the `_styles.css` seems redundant when `.css` is used anyways, I think css styling is implied

File: package.json (hunk 1 above)
Comment (mihok) [inline package.json]: `style-loader`, and `css-loader` is missing from the devDependencies

File: test/script.test.js [hunk 3]
```diff
@@ -20,8 +20,8 @@ io.connect = jest.fn(() => socket);
 //  components reception)
 //    - UI_OPEN
 //    - UI_CLOSE
-//    - UI_STYLE_MESSANGER
-//    - UI_STYLE_MESSANGER
+//    - UI_STYLE_MESSENGER
+//    - UI_STYLE_MESSENGER
```
Comment (mihok) [inline test/script.test.js]: Hahaha, horay english

File: webpack.config.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 var webpack = require('webpack');
 var development = process.env.NODE_ENV !== "production";
+var path = require('path');
 
 var PATHS = {
-  BUILD: __dirname + '/dist',
+  BUILD: path.join(__dirname, '..', 'example_site/public'),
```
Comment (mihok) [inline webpack.config.js]: This is dangerous in that your referencing a folder outside of the scope of this code repo, you should use a symlink in your example_site instead and keep this to `/dist`

File: src/components/Input/Input_styles.css (hunk 2 above)
Comment (teesloane) [inline src/components/Input/Input_styles.css]: I will rename these!

File: webpack.config.js (hunk 4 above)
Comment (teesloane) [inline webpack.config.js]: hmm. didn't know that -  not sure how I would do the sym link - might have to go over this with you to learn the best way to do it.

**Comment by teesloane** [comment]
Notes:

I'm not sure the best way to make this PR. Perhaps switching to a new branch on the original upstream (here) would be best. Hard to say.

**Comment by teesloane** [comment]
update:
I moved the tests from the original `script.test.js` file into seperate tests per component folder. This is similar to how things are done on the `create-react-app` scaffold I believe, but we can figure out a structure whenever. It's a big ugly because now `__snapshots__` get added to the component folder.



Some stuff going green!

**Pull request merged**